api/institutions/authentication.py

```diff
@@ -54,10 +54,10 @@ def authenticate(self, request):
         user, created = get_or_create_user(fullname, username, reset_password=False)
 
         if created:
-            user.given_name = provider['user'].get('givenName')
-            user.middle_names = provider['user'].get('middleNames')
-            user.family_name = provider['user'].get('familyName')
-            user.suffix = provider['user'].get('suffix')
+            user.given_name = provider['user'].get('givenName') or user.given_name
```

Comment:
@icereval  Looks good. 🔥 

Currently we don't have any issue with `fullname` being `None`, which will happen for OK State University. (See https://github.com/CenterForOpenScience/osf.io/pull/6754/commits/69e04b5711e3886ff4cfe578d4e095c574b1b7ef) I can fix the conflicts when this gets merged. Mine code are not as concise as @caseyrollins's and I forgot to add `middle_names` and `suffix` which are also guessed by `update_guessed_names`: [link to the code](https://github.com/CenterForOpenScience/osf.io/blob/develop/framework/auth/core.py#L616).